runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/Info.java

```diff
@@ -72,8 +87,65 @@ public void reset()
         }
     }
 
+    private int readWorkers()
+    {
+        try
+        {
+            byte[] bytes = Files.readAllBytes(info);
+            ByteBuffer byteBuf = ByteBuffer
+                .wrap(bytes, Long.BYTES, Integer.BYTES)
+                .order(nativeOrder());
+            return byteBuf.getInt();
+        }
+        catch (IOException ex)
+        {
+            System.out.printf("Error: %s is not readable\n", info);
+            return 0;
+        }
+    }
+
     @Override
     public void close() throws Exception
     {
     }
+
+    public static final class Builder
+    {
+        private Path path;
+        private int workerCount;
+        private boolean readonly;
+
+        public Builder path(
+            Path path)
+        {
+            this.path = path;
+            return this;
+        }
+
+        public Builder workerCount(
+            int workerCount)
+        {
+            this.workerCount = workerCount;
+            return this;
+        }
+
+        public Builder readonly(
+            boolean readonly)
+        {
+            this.readonly = readonly;
+            return this;
+        }
+
+        public Info build()
+        {
+            if (readonly)
+            {
+                return new Info(path);
+            }
+            else
+            {
+                return new Info(path, workerCount);
+            }
```

Comment:
Please do the work for `readonly` vs `!readonly` here in the builder, instead of calling from the constructor.
We follow a similar approach in some of the other *Layout classes that support optional `readonly` behavior.

Reply:
fixed

Reply:
I meant do the work here inline, no need for those other methods.

Reply:
Pls clarify.

This is the current code:
```
        public Info build()
        {
            Info info = new Info(path);
            if (readonly)
            {
                info.readWorkerCount();
            }
            else
            {
                info.reset(workerCount);
            }
            return info;
        }
```

Do you mean that I should inline the `readWorkerCount` and the `reset` methods from the `Info` class to the `build` method of the `Info.Builder` class? 🤔 Pls LMK.

Reply:
Yes

Reply:
OK, done.